src/Client/ClientBase.cpp

```diff
@@ -143,6 +145,78 @@ namespace ProfileEvents
 namespace
 {
 constexpr UInt64 THREAD_GROUP_ID = 0;
+
+void handleTruncateMode(DB::ASTQueryWithOutput * query_with_output, const String & out_file, String & query)
+{
+    if (!query_with_output->is_outfile_truncate)
+        return;
+
+    // Skip atomic rename for special files
```

Comment:
Please use `///` for all single-line comments to follow [our coding style](https://clickhouse.com/docs/development/style#comments)

>8. Single-line comments begin with three slashes: /// and multi-line comments begin with /**. These comments are considered "documentation".